### ISSUE TITLE
(BUG) Cloned campaign is broken and events nodes have to be reconnected manually

### ISSUE THREAD

**Issue by AlbanL74fr** [mautic/mautic#7127]
## Bug Description

I have a campaign that is very simple, but can't be clone successfully.
As soon as I clone and save (the clone), then an error page occurs. I need to hit return and go to the campaign builder to "reconnect" the events

| Q   | A
| --- | ---
| Mautic version | 2.15.0
| PHP version | 7.1.25
| Browser | 

### Steps to reproduce
1. Create a new campaign such as in the screenshot below

![image](https://user-images.githubusercontent.com/23435414/51070735-d95abd80-1678-11e9-94e8-5cd16d778f72.png)

2. Clone the campaign
3. Save the clone

The campaign is saved but an error page is shown 

![image](https://user-images.githubusercontent.com/23435414/51070758-1921a500-1679-11e9-96c3-9d84d611d1b5.png)

Go back to the clone and see the campaign builder :

![image](https://user-images.githubusercontent.com/23435414/51070774-4ec68e00-1679-11e9-9527-8f0c08b2eaf6.png)

 
### Log errors

```
[2019-01-12 14:36:24] mautic.CRITICAL: Uncaught PHP Exception Doctrine\DBAL\Exception\UniqueConstraintViolationException: "An exception occurred while executing 'INSERT INTO campaign_form_xref (campaign_id, form_id) VALUES (?, ?)' with params [15, 10]:  SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '15-10' for key 'PRIMARY'" at /var/www/html/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php line 66 {"exception":"[object] (Doctrine\\DBAL\\Exception\\UniqueConstraintViolationException(code: 0): An exception occurred while executing 'INSERT INTO campaign_form_xref (campaign_id, form_id) VALUES (?, ?)' with params [15, 10]:\n\nSQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '15-10' for key 'PRIMARY' at /var/www/html/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php:66, Doctrine\\DBAL\\Driver\\PDOException(code: 23000): SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '15-10' for key 'PRIMARY' at /var/www/html/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOStatement.php:107, PDOException(code: 23000): SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '15-10' for key 'PRIMARY' at /var/www/html/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOStatement.php:105)"} []
```

[//]: # ( Invisible comment:
Please check for related errors in the latest log file in [mautic root]/app/log/ and/or the web server's logs and post them here. Be sure to remove sensitive information if applicable. )

**Comment by slatapi** [comment]
Just got this today.

**Comment by ISKCONonlineadmin** [comment]
We're also experiencing this, but with a pretty complex campaign.

**Comment by thomeudt** [comment]
Happens to us with every campaign we are cloning, and it's considerable extra work to create each campaign from scratch.

**Comment by Corsnec** [comment]
I have the same problem.

**Comment by Aftershok** [comment]
Same here. Surprised I haven't come across this before.

**Comment by luizcarlosfaria** [comment]
I have same issue.

**Comment by florentpetitjean** [comment]
Got the same issue, it seems happening only on campaign starting with a form

**Comment by dkossako** [comment]
Also present in 1.15.1

`[2019-06-11 11:26:37] mautic.CRITICAL: Uncaught PHP Exception Doctrine\DBAL\Exception\UniqueConstraintViolationException: "An exception occurred while executing 'INSERT INTO campaign_form_xref (campaign_id, form_id) VALUES (?, ?)' with params [15, 19]:  SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '15-19' for key 'PRIMARY'" at /var/www/html/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php line 66 {"exception":"[object] (Doctrine\\DBAL\\Exception\\UniqueConstraintViolationException(code: 0): An exception occurred while executing 'INSERT INTO campaign_form_xref (campaign_id, form_id) VALUES (?, ?)' with params [15, 19]:\n\nSQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '15-19' for key 'PRIMARY' at /var/www/html/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php:66, Doctrine\\DBAL\\Driver\\PDOException(code: 23000): SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '15-19' for key 'PRIMARY' at /var/www/html/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOStatement.php:107, PDOException(code: 23000): SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '15-19' for key 'PRIMARY' at /var/www/html/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOStatement.php:105)"} []`

**Comment by kuzmany** [comment]
Let's test: https://github.com/mautic/mautic/pull/7642

**Comment by dkossako** [comment]
Do we have any ETA for this fix?

**Comment by npracht** [comment]
Merged in 2.15.3
